container/writer.go

```diff
@@ -45,8 +45,10 @@ type Writer struct {
   Create a new Writer wrapping the provided io.Writer with the given Codec and number of records per block.
   The Writer will lazily write the container file header when WriteRecord is called the first time.
   You must call Flush on the Writer before closing the underlying io.Writer, to ensure the final block is written.
+  Optionally, pass in a schema string to ensure that the header is written even if no records are written. This
+  is required to produce valid empty Avro container files.
 */
-func NewWriter(writer io.Writer, codec Codec, recordsPerBlock int64) (*Writer, error) {
+func NewWriter(writer io.Writer, codec Codec, recordsPerBlock int64, schema ...string) (*Writer, error) {
```

Comment:
Instead of making this method variadic, you can go ahead and make `schema` required. The package is versioned with gopkg.in, I'm not worried about breaking changes on master.